lib/danger_plugin.rb

```diff
@@ -39,33 +39,67 @@ def lint_files
         return
       end
 
-      swiftlint_command = ""
-      swiftlint_command += "cd #{directory} && " if directory
-      swiftlint_command += "swiftlint lint --quiet --reporter json"
-      swiftlint_command += " --config #{config_file}" if config_file
-
-      require 'json'
-      result_json = JSON.parse(`(#{swiftlint_command})`).flatten
-
-      # Convert to swiftlint results
-      warnings = result_json.select do |results| 
-        results['severity'] == 'Warning'
-      end
-      errors = result_json.select do |results| 
-        results['severity'] == 'Error' 
-      end
-
-      message = ''
-
-      # We got some error reports back from swiftlint
-      if warnings.count > 0 || errors.count > 0
-        message = "### SwiftLint found issues\n\n"
+      require 'tempfile'
+      Tempfile.open('.swiftlint_danger.yml') do |temp_config_file|
```

Comment:
Ensuring that temporary files are properly cleaned up is important, using this block based syntax seems like a good way to ensure it's always cleaned up. The disadvantage is that it adds a bit of complexity to the method.